src/consumption.cpp

```diff
@@ -1095,6 +976,134 @@ void Character::modify_morale( item &food, const int nutr )
         add_morale( MORALE_FOOD_GOOD, fun.first, fun.second, morale_time, morale_time / 2, false,
                     food.type );
     }
+
+    bool spoiled = food.rotten();
+
+    // Morale bonus for eating unspoiled food at a nearby table
+    if( !spoiled && !food.has_flag( flag_ALLERGEN_JUNK ) ) {
+        bool has_table_nearby = false;
+        bool has_chair_nearby = false;
+        for( const tripoint &pt : g->m.points_in_radius( pos(), 1 ) ) {
+            if( g->m.has_flag_furn( flag_FLAT_SURF, pt ) || g->m.has_flag( flag_FLAT_SURF, pt ) ||
+                ( g->m.veh_at( pt ) && ( g->m.veh_at( pt )->vehicle().has_part( "KITCHEN" ) ||
+                                         g->m.veh_at( pt )->vehicle().has_part( "FLAT_SURF" ) ) ) ) {
+                has_table_nearby = true;
+            }
+            if( g->m.has_flag_furn( flag_CAN_SIT, pt ) || g->m.has_flag( flag_CAN_SIT, pt ) ||
+                ( g->m.veh_at( pt ) && ( g->m.veh_at( pt )->vehicle().has_part( "SEAT" ) ) ) ) {
+                has_chair_nearby = true;
```

Comment:
This can allow early exit when you have both `has_table_nearby` and `has_chair_nearby`.

Reply:
I am guessing you mean something like `if(has_table_nearby) {break;}` ? Assuming I plan to go ahead and factor out the whole `has_table_nearby` operation, I think this will become a moot point, so I've left it alone for now.